### PR TITLE
Request to review and merge quickstart demo using Hibernate in AS7

### PR DESCRIPTION
**Pull request by madhumita12**

### CODE REVIEW

**Comment by madhumita12** [comment]
However I am yet to modify the table/entity name with reference to Sande's mail yesterday

**Comment by sgilda** [comment]
Do you plan on doing this soon? if so, I'll wait until you do that and issue another pull. :-)

File: hibernatedemo/pom.xml [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+      </dependency>
+      
+      <dependency>
+         <groupId>org.hibernate</groupId>
+         <artifactId>hibernate-entitymanager</artifactId>
+         <version>4.0.0.CR6</version>
+         <scope>provided</scope>
+      </dependency>
+      
+      <dependency>
```
Comment (pmuir) [inline hibernatedemo/pom.xml]: I think we need a comment on this dependency, given it's basically the key part of this quickstart!

**Comment by pmuir** [comment]
Hi Madhumita,

Sorry, it wasn't clear from the original description, but in this example we should include instructions on how to use Hibernate APIs in AS 7. Can you update accordingly?

**Comment by madhumita12** [comment]
Pete sure I will update the readme accordingly.I would like to confirm that this example uses mixed JPA and Hibernate Session (Native API) using hibernate 4 and JPA shipped with AS7 and is not pure Hibernate example(using only native api).

**Comment by pmuir** [comment]
Sounds good.

**Comment by sgilda** [comment]
Could we rename this one to hibernate4-demo or hibernate4 (depending on what happens with naming of the hibernate 3 demo?

**Comment by madhumita12** [comment]
yes I am renaming to hibernate4-demo as suggested by you.

**Comment by sgilda** [comment]
Great! Thanks!

**Comment by sgilda** [comment]
Per Pete's comment on the other hibernate quickstart, you can drop the -demo and just make it hibernate4.

**Comment by madhumita12** [comment]
Sande/Pete,

Please could you take a look again and suggest in case I missed anything as per earlier comments....

**Comment by sgilda** [comment]
Looks great! Tested and works fine. 
My only comments, which are similar to what I said for the hibernate3 demo:
-- The URL for this quickstart is: http://localhost:8080/hibernate4/
   The URL for hibernate3 is: http://localhost:8080/jboss-as-hibernate3/
It would be nice if they were consistent. Most of the quickstart contexts start with "jboss-as-" it might be better to change this one, but I don't feel strongly either way as long as they're consistent.  Pete, do you have a preference?
-- This is just a suggestion: Comment the code differences between Hibernate 3 and Hibernate 4 in the classes and add an overview of the differences the README file along with a list of files they should compare between the 2 examples.

**Comment by madhumita12** [comment]
Thanks Sande
I think I should change the url to http://localhost:8080/jboss-as-hibernate4/ in compliance with other quickstarts.
Will add another round of comments specifying differences.

**Comment by sgilda** [comment]
That sounds good to me. Thanks!

**Comment by madhumita12** [comment]
Sande/Pete,

```
 I also removed the exclusions added to hibernate-core dependency as they are redundant for AS7 but was required with the current build of EAP6 and is getting fixed.You might like to build this once more on your systems for verification that things are working fine.
```

**Comment by sgilda** [comment]
Reviewed for comments and readme. The new context looks good and is in-line with the other quickstarts.
Pete, when you've done a technical review, I think this will be good to go.

File: hibernate4/readme.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+The application this project produces is designed to be run on a JBoss AS 7 or EAP 6. 
+The following instructions target JBoss AS 7, but they also apply to JBoss EAP 6.
+ 
+With the prerequisites out of the way, you're ready to build and deploy.
+
+
+Adding correct Dependencies
+---------------------------
+
+AS7 (or EAP 6) provides JPA API and several ORM implementations like OpenJPA,Hibernate4 etc.
```
Comment (pmuir) [inline hibernate4/readme.md]: I don't think JBoss AS 7 (not AS7) provides OpenJPA?!?

**Comment by pmuir** [comment]
Minor comments only :-) Otherwise great!

**Comment by madhumita12** [comment]
We have some compatibility tests in JBOSS AS7 here :
https://github.com/jbossas/jboss-as/tree/master/testsuite/compat/src/test/java/org/jboss/as/test/compat/jpa
I think Jboss AS7 provides OPENJPA as persistence provider 
But we do not support it in EAP6 .

**Comment by pmuir** [comment]
I talked to Jason Greene, he said that OpenJPA is supported, but not provided. Can we change the language accordingly?

**Comment by madhumita12** [comment]
Sure...btw in EAP6 afaik we only support JPA with Hibernate and do not provide any additional support for OpenJPA.
I am wondering if I should remove (or EAP 6) from related text above.Please let me know.

**Comment by sgilda** [comment]
Madhumita, I'm unsure from your comment above whether you have made the language changes Pete described above? Let me know when you are ready.

**Comment by sgilda** [comment]
Sorry for the delay in looking at this one. I haven't had time to test it yet, but I did notice that your pom.xml file has some wrong versions. 
Can you change this section as follows (not the version and name):
   <groupId>org.jboss.as.quickstart</groupId>
   <artifactId>jboss-as-hibernate4</artifactId>
   <version>7.0.2-SNAPSHOT</version>
   <packaging>war</packaging>
   <name>JBoss AS Quickstarts: Hibernate 4 application</name>
Some of the other versions in the pom.xml file are also incorrect, for example, javaee6.web.spec.version. I've created a page that lists some of them here: https://docs.jboss.org/author/display/AS71/Maven+POM+Versions+Checklist
You can also look at some of the other quickstarts in https://github.com/jbossas/quickstart/ since they've all been fixed to work now.
Sorry about that, but we've run into a lot of issues testing and merging the quickstarts to EAP because of incorrect version numbers in the pom.xml.
Thanks,
Sande

**Comment by sgilda** [comment]
Hi Madhumita,
I've created a page with the correct version numbers to use in the POM files: 
    https://docs.jboss.org/author/display/AS71/Maven+POM+Versions+Checklist
It should help take the guesswork out of this. :-)
Thanks, Sande

**Comment by madhumita12** [comment]
Hi Sande,

```
        I am on PTO in India and do not have regular access to office network and hence sorry about the delayed response.
```

I have gone through above link and also existing quickstarts and corrected the versions in pom.xml.
I also tested this with AS 7.1.0.CR1 and it runs fine.
Please let me know if anything is missing and I will try to get back when I get a chance to check my emails again.

**Comment by sgilda** [comment]
Hi Madhumita,
I did notice one difference in the POM. For hibernate-core, you have 4.0.0.Final, but the chart https://docs.jboss.org/author/display/AS71/Maven+POM+Versions+Checklist is showing 3.6.8.Final. Is the chart incorrect?
Thanks, Sande

**Comment by madhumita12** [comment]
Hi Sande,

```
          We are no more shipping hibernate 3 as part of AS7 but package only hibernate 4.
```

Hence we have two separate quickstarts for hibernate
1. hibernate4 showing usage of hibernate4 shipped with as7
2. hibernate 3 showing usage of hibernate 3 jars bundled so that they can be used with as7

You may check the exact versions of hibernate jars packed within $JBOSS_HOME/modules/org/hibernate/main
hibernate-core-4.0.0.Final
hibernate-entitymanager-4.0.0.Final etc

As for Hibernate 3 user can use any version of hibernate 3 as per his discretion as AS7 supports hibernate3 even though we do not ship it.

Thanks,
Madhumita

**Comment by sgilda** [comment]
Hi Madhumita,
Would you mind making the following changes to your README files:
- Could you change the title to: 
      hibernate4: How to Use Hibernate 4 in an Application
- Could you also add yourself as Author in the line below the title so you get credit? You'll see that in all the README files now.
      Author: Madhumita Sadhukhan  

The only difference I see now is the version for hibernate-core. I think we might have updated this since you issued your pull. You have 4.0.0.Final and the checklist and Paul's doc show 4.0.1.Final. If we run scripts to modify the version for EAP, this might not get picked up.  Sorry about that.

Other than that, this one is ready to go. :-)

**Comment by sgilda** [comment]
Modified the hibernate-core version in the pom.xml file. Renamed and formatted the README files. 
This pull is being replaced by pull 82. When that one is verified and processed, we can close this one.

**Comment by madhumita12** [comment]
Hi Sande,

```
      I just checked the hibernate-core version in AS jboss-as-7.1.0.CR1b and the hibernate-core version is still 4.0.0.Final.I also tested this application against jboss-as-7.1.0.CR1b using 4.0.0.Final for hibernate-core and it runs fine.
```

I have made the changes in readme as suggested but I am not yet sure of the version to be used for hibernate-core.

**Comment by madhumita12** [comment]
Hi Sande,

```
            Sorry there seem to be some parallel updates.I saw your comment after my commit
```

**Comment by pmuir** [comment]
Merged pull 82. Closing this one.

**Pull request closed without merging**